### PR TITLE
Use docker instead of docker-compose for CI

### LINKED ISSUE #40: CI docker service connection issues
There seems to be an issue with connecting to the docker services in the "real backend" CI job.
See also SINTEF/oteapi-optimade#34 for a similar issue and a possible solution.

### PR DESCRIPTION
**Pull request by CasperWA**
# Description:

Use docker directly instead of docker-compose in CI test for running PyTest with a "real" backend.

There have been continued connection issues with using docker-compose in CI - see #40.

Fixes #40 

Also, remove the CI docker-compose file, which is now never used.

Furthermore, this PR disables import errors when running `pylint` for files under `tests`, since import errors are raised for the local package when using the latest `pylint` version.

## Type of change:

- [X] Bug fix.
- [ ] New feature.
- [ ] Documentation update.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: .github/workflows/ci_tests.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     - name: Set up Python 3.9
       uses: actions/setup-python@v4
       with:
-        python-version: 3.9
```
Comment (daniel-sintef) [inline .github/workflows/ci_tests.yml:22]: Why the change from no quotes to with quotes?

File: .github/workflows/ci_tests.yml [hunk 2]
```diff
@@ -62,7 +62,7 @@ jobs:
       run: pylint --rcfile=pyproject.toml --ignore-paths=tests/ --extension-pkg-whitelist='pydantic' *.py otelib
 
     - name: Run pylint - tests
-      run: pylint --rcfile=pyproject.toml --extension-pkg-whitelist='pydantic' --disable=import-outside-toplevel,redefined-outer-name,duplicate-code tests
+      run: pylint --rcfile=pyproject.toml --extension-pkg-whitelist='pydantic' --disable=import-outside-toplevel,redefined-outer-name,duplicate-code,import-error tests
```
Comment (daniel-sintef) [inline .github/workflows/ci_tests.yml:65]: Is this where the error was, we needed the 'import-error'?

File: .github/workflows/ci_tests.yml (hunk 1 above)
Comment (CasperWA) [reply .github/workflows/ci_tests.yml:22]: Because this is the correct way to assign this value. One will find that this is true when trying to use `3.10`. This will fail unless wrapped in quotes.

File: .github/workflows/ci_tests.yml [hunk 3]
```diff
@@ -131,12 +131,19 @@ jobs:
   pytest-real-backend:
     runs-on: ubuntu-latest
     name: pytest-real backend
-    # timeout-minutes: 15
 
     env:
       PORT: 8080
       OTEAPI_prefix: "/api/v1"
 
+    services:
```
Comment (daniel-sintef) [inline .github/workflows/ci_tests.yml:139]: Or I guess maybe the error was this service wasn't there?

File: .github/workflows/ci_tests.yml [hunk 4]
```diff
@@ -155,9 +162,17 @@ jobs:
 
     - name: Run OTE server
       run: |
-        docker-compose -f .github/utils/docker-compose.yml up &
+        docker pull ghcr.io/emmc-asbl/oteapi:${DOCKER_OTEAPI_VERSION}
```
Comment (daniel-sintef) [inline .github/workflows/ci_tests.yml:165]: Why the change from a separate docker-compose?

File: README.md [hunk 5]
```diff
@@ -143,8 +143,8 @@ OTELib is released under the [MIT license](LICENSE) with copyright &copy; SINTEF
 
 OTEAPI Core has been supported by the following projects:
 
-* __OntoTrans__ (2020-2024) that receives funding from the European Union’s Horizon 2020 Research and Innovation Programme, under Grant Agreement no. 862136.
```
Comment (daniel-sintef) [inline README.md:146]: Any reason to change from __formating__ to **formatting** ?

File: .github/workflows/ci_tests.yml (hunk 2 above)
Comment (CasperWA) [reply .github/workflows/ci_tests.yml:65]: This is a separate error. I just added a sentence about it in the PR description. It needs to be added here as well as in the `pre-commit` configuration file.

**Review by daniel-sintef — APPROVED** [review]

File: .github/workflows/ci_tests.yml [hunk 6]
```diff
@@ -131,12 +131,19 @@ jobs:
   pytest-real-backend:
     runs-on: ubuntu-latest
     name: pytest-real backend
-    # timeout-minutes: 15
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply .github/workflows/ci_tests.yml:134]: This was indeed just a comment before - but sure, I can leave it in for someone to un-comment it in the future? Or you want to add it now?

File: .github/workflows/ci_tests.yml (hunk 3 above)
Comment (CasperWA) [reply .github/workflows/ci_tests.yml:139]: The error was the use of `docker-compose` in this job, which the GitHub Actions CI framework has some issues with, apparently. Instead I now use `docker`.

File: .github/workflows/ci_tests.yml (hunk 4 above)
Comment (CasperWA) [reply .github/workflows/ci_tests.yml:165]: This is the error, and hence the fix.

File: README.md (hunk 5 above)
Comment (CasperWA) [reply README.md:146]: Only to satisfy the markdown-linter I have locally, which wants the formatting to be the same throughout as much as possible. I.e., if `**` is previously used, it should be used throughout. It doesn't prefer one type in favor of another, just consistency. So, if you prefer it one way or another I don't mind using `__` instead of `**` throughout?
